packages/core/src/queue/event-queue.ts

```diff
@@ -274,16 +279,18 @@ export abstract class CoreEventQueue<
       if (temp instanceof CoreContext) {
         ctx = temp
       }
+
+      this.emit('message_enriched', ctx, beforeWare)
```

Comment:
We probably want to get into the habit of writing some tests for these emitted events since we are moving into  treating them as a first class API.